packages/aura/src/components/side-nav.css

```diff
@@ -24,67 +23,43 @@ vaadin-side-nav-item:not([disabled])::part(content):hover {
+  --_accent: light-dark(
+    oklch(from var(--aura-accent-light) calc(l + 0.2) c h / min(0.3, c / 2)),
+    oklch(from var(--aura-accent-dark) calc(l - 0.2) c h / min(0.3, c / 2))
+  );
+  --_accent-border: light-dark(
+    oklch(from var(--aura-accent-light) l calc(c / 2) h / calc(min(0.15, 0.05 + c / 2) + 0.1 * var(--aura-contrast))),
+    oklch(from var(--aura-accent-dark) l calc(c / 2) h / calc(min(0.15, 0.05 + c / 2) + 0.1 * var(--aura-contrast)))
+  );
+  --vaadin-side-nav-item-background: linear-gradient(var(--_accent), var(--_accent)) var(--aura-surface) padding-box;
+  --vaadin-side-nav-item-color: var(--aura-accent-text);
+  --vaadin-side-nav-item-border-color: var(--_accent-border);
 }
 
-/* Contrast variant */
+/* Filled variant */
 
-vaadin-side-nav[theme~='contrast'] vaadin-side-nav-item[current]::part(content) {
-  color-scheme: var(--_side-nav-current-item-color-scheme, dark);
-  background: var(--aura-surface-solid);
-  border-color: transparent;
+vaadin-side-nav[theme~='filled'] vaadin-side-nav-item[current]::part(content) {
+  --vaadin-side-nav-item-background: var(--aura-accent-color) border-box;
+  --vaadin-side-nav-item-color: var(--aura-accent-contrast);
   outline-offset: 2px;
 }
 
-vaadin-side-nav[theme~='contrast'] vaadin-side-nav-item[current] > :not([slot='children']) {
-  --aura-red: var(--vaadin-text-color);
-  --aura-red-text: var(--vaadin-text-color);
-  --aura-orange: var(--vaadin-text-color);
-  --aura-orange-text: var(--vaadin-text-color);
-  --aura-yellow: var(--vaadin-text-color);
-  --aura-yellow-text: var(--vaadin-text-color);
-  --aura-green: var(--vaadin-text-color);
-  --aura-green-text: var(--vaadin-text-color);
-  --aura-blue: var(--vaadin-text-color);
-  --aura-blue-text: var(--vaadin-text-color);
-  --aura-purple: var(--vaadin-text-color);
-  --aura-purple-text: var(--vaadin-text-color);
-}
-
-@container style(--aura-color-scheme: dark) {
-  [slot='drawer'] {
-    --_side-nav-current-item-color-scheme: light;
-  }
-}
-
-@container style(--aura-content-color-scheme: dark) {
-  vaadin-app-layout > :not([slot]) {
-    --_side-nav-current-item-color-scheme: light;
-  }
-}
-
-@container style(--aura-content-color-scheme: light) {
-  vaadin-app-layout > :not([slot]) {
-    --_side-nav-current-item-color-scheme: dark;
-  }
-}
-
-@media (prefers-color-scheme: dark) {
-  @container style(--aura-content-color-scheme: light dark) {
-    vaadin-app-layout > :not([slot]) {
-      --_side-nav-current-item-color-scheme: light;
-    }
-  }
-
-  @container style(--aura-color-scheme: light dark) {
-    [slot='drawer'] {
-      --_side-nav-current-item-color-scheme: light;
-    }
-  }
-}
-
-/* TODO bug in component/base styles: hide label part if no label is provided */
-vaadin-side-nav:not(:has([slot='label']))::part(label) {
-  display: none;
+/* Badges and other content that uses the palette inside a filled nav item */
+vaadin-side-nav[theme~='filled'] vaadin-side-nav-item[current] > :not([slot='children']) {
```

Comment:
Should we change selector to ensure this doesn't affect tooltips in `slot="tooltip"` added in #10008?

Reply:
Yeah, should exclude tooltips as well.